### PR TITLE
8302512: Update IANA Language Subtag Registry to Version 2023-02-14

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
Incorporate the latest IANA language subtag registry definition (2023-02-14).

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8302512](https://bugs.openjdk.org/browse/JDK-8302512): Update IANA Language Subtag Registry to Version 2023-02-14

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12699/head:pull/12699` \
`$ git checkout pull/12699`

Update a local copy of the PR: \
`$ git checkout pull/12699` \
`$ git pull https://git.openjdk.org/jdk pull/12699/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12699`

View PR using the GUI difftool: \
`$ git pr show -t 12699`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12699.diff">https://git.openjdk.org/jdk/pull/12699.diff</a>

</details>

### CODE REVIEW

File: test/jdk/java/util/Locale/LanguageSubtagRegistryTest.java [hunk 1]
```diff
@@ -24,9 +24,9 @@
 /*
  * @test
  * @bug 8040211 8191404 8203872 8222980 8225435 8241082 8242010 8247432
- *      8258795 8267038
+ *      8258795 8267038 8287180 8302512
  * @summary Checks the IANA language subtag registry data update
- *          (LSR Revision: 2022-08-08) with Locale and Locale.LanguageRange
+ *          (LSR Revision: 2023-2-14) with Locale and Locale.LanguageRange
```
Comment (naotoj) [inline test/jdk/java/util/Locale/LanguageSubtagRegistryTest.java:29]: Add prepending `0` to the month to keep the YYYY-MM-DD format.

**Review by naotoj — COMMENTED** [review]
Looks good. Please update the copyright year for the test changes.

File: test/jdk/java/util/Locale/LanguageSubtagRegistryTest.java (hunk 1 above)
Comment (justin-curtis-lu) [reply test/jdk/java/util/Locale/LanguageSubtagRegistryTest.java]: Thanks for catching, fixed

**Review by naotoj — APPROVED** [review]

**Pull request closed without merging**